addons/project/views/project_task_views.xml

```diff
@@ -113,11 +112,10 @@
                 <filter name="my_tasks" position="before">
                     <field string="Properties" name="task_properties"/>
                 </filter>
-                <filter name="inactive" position="before">
+                <filter name="closed_on" position="after">
```

Comment:
Be careful that after `<filter string="Closed On" name="closed_on" ...>`, there is a `<separator/>` (see https://github.com/odoo/odoo/blob/master/addons/project/views/project_task_views.xml#L33)
You may need to take that into account

Reply:
@shup-odoo is it really to add an extra separator since there is already a separator after close_on filter as Maxime mentioned?

Reply:
Indeed the separator is needed here to separate `message_needaction` and `show_subtasks`
FYI - The new filter `show_subtasks` was added in this commit, so I made the changes accordingly to it.